File: Class_4/OOP/go/greeter/Greeters/popup_greeter.go
package greeters

import "github.com/sqweek/dialog"

const title string = "Greetings!"

type PopupGreeter struct {
	name string
}

func (_ PopupGreeter) Greet(name string) {
	dialog.Message("%s", "Greet you"+name).Title(title).Info()
}

File: Class_4/OOP/java/org/greeter/Greeters/PopupGreeter.java
package org.greeter.Greeters;

import javax.swing.JOptionPane;

public class PopupGreeter implements Greeter {
    @Override
    public void greet(String name) {
        this.infoBox("Greet you " + name, "Greetings!");
    }

    public void infoBox(String infoMessage, String titleBar) {
        JOptionPane.showMessageDialog(null, infoMessage, "InfoBox: " + titleBar, JOptionPane.INFORMATION_MESSAGE);
    }
}